examples/cpp_classification/classification.cpp

```diff
@@ -0,0 +1,255 @@
+#include <caffe/caffe.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <iosfwd>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+using namespace caffe;  // NOLINT(build/namespaces)
+using std::string;
+
+/* Pair (label, confidence) representing a prediction. */
+typedef std::pair<string, float> Prediction;
+
+class Classifier {
+ public:
+  Classifier(const string& model_file,
+             const string& trained_file,
+             const string& mean_file,
+             const string& label_file);
+
+  std::vector<Prediction> Classify(const cv::Mat& img, int N = 5);
+
+ private:
+  void SetMean(const string& mean_file);
+
+  std::vector<float> Predict(const cv::Mat& img);
+
+  void WrapInputLayer(std::vector<cv::Mat>* input_channels);
+
+  void Preprocess(const cv::Mat& img,
+                  std::vector<cv::Mat>* input_channels);
+
+ private:
+  std::auto_ptr<Net<float> > net_;
```

Comment:
auto_ptr is deprecated in C++11. It's quite plausible that caffe at some point would move to using C++11/C++14 (and hopefully before C++17 implementations are available to us!)

I'm using a caffe fork that's CPU only that I'm building with C++11. I'd like to use your code as my "application" layer, hence my request to use boost's smart pointers instead of auto_ptr.  

I understand that you don't want to build out dependencies but given that boost is being used anyway, my preference would be to use one of its smart pointer. It's almost tragic that C++11